#! /usr/bin/env python
#Autogenerated V4L Dynamic Control

PACKAGE='v4r_ellipses'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("show_camera_image", bool_t, 0, "Show camera image", True)
gen.add("debug", bool_t, 0, "Shows additional debug information", False)
gen.add("threshold_edge_detection", int_t, 0, "edge detection", 150, 0, 255)
gen.add("threshold_contour_min_points", int_t, 0, "edge detection", 20, 10, 255)
gen.add("threshold_polygon", int_t, 0, "epsylon to approximate polygon", 8, 3, 16)
gen.add("filter_convex", bool_t, 0, "rings only", True)
gen.add("threshold_rotated_rect_ratio", double_t, 0, "ring cirlce ratio", 0.02, 0, 1)
gen.add("filter_rings", bool_t, 0, "rings only", True)
gen.add("threshold_ellipse_mean", double_t, 0, "ring cirlce ratio", 0.05, 0, 0.5)
gen.add("filter_ellipse_mean_sample_steps", int_t, 0, "sample steps", 1, 0, 20)
gen.add("threshold_ring_center", double_t, 0, "ring center offsets", 0.05, 0, 0.5)
gen.add("threshold_ring_ratio", double_t, 0, "ring cirlce ratio", 0.1, 0, 1.0)

exit(gen.generate(PACKAGE, "v4r_ellipses", "EllipsesDetection"))
